operator/controllers/common/otel_utils_test.go

```diff
@@ -0,0 +1,227 @@
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/metric/instrument/asyncfloat64"
+	"go.opentelemetry.io/otel/metric/instrument/asyncint64"
+	"go.opentelemetry.io/otel/metric/instrument/syncfloat64"
+	"go.opentelemetry.io/otel/metric/instrument/syncint64"
+	"google.golang.org/grpc"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestGetOTelTracerProviderOptions(t *testing.T) {
+	listener, err := net.Listen("tcp", ":9000")
+	if err != nil {
+		panic(err)
+	}
+
+	s := grpc.NewServer()
+	go func() {
+		err := s.Serve(listener)
+		if err != nil {
+			panic(err)
+		}
+	}()
+
+	type args struct {
+		oTelCollectorUrl string
+	}
+	tests := []struct {
+		name            string
+		args            args
+		wantArrayLength int
+		wantErr         bool
+	}{
+		{
+			name: "Test with no URL",
+			args: args{
+				oTelCollectorUrl: "",
+			},
+			wantArrayLength: 2,
+		},
+		{
+			name: "Test with wrong URL",
+			args: args{
+				oTelCollectorUrl: "error-url",
+			},
+			wantArrayLength: 0,
+			wantErr:         true,
+		},
+		{
+			name: "Test with URL",
+			args: args{
+				oTelCollectorUrl: "localhost:9000",
+			},
+			wantArrayLength: 3,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			//TODO also test underline return
+			got, _, err := GetOTelTracerProviderOptions(tt.args.oTelCollectorUrl)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("GetOTelTracerProviderOptions() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if len(got) != tt.wantArrayLength {
+				t.Errorf("GetOTelTracerProviderOptions() got length = %v, wantArrayLength %v", got, tt.wantArrayLength)
+
+			}
+		})
+	}
+}
+
+func TestSetUpKeptnMeters(t *testing.T) {
+	fakeAsyncIntTracerProvider := &fakeasync.ITracerProviderAsyncInt64Mock{
+		GaugeFunc: func(name string, opts ...instrument.Option) (asyncint64.Gauge, error) {
+			return nil, errors.New("some error")
+		},
+	}
+	fakeAsyncFloatTracerProvider := &fakeasync.ITracerProviderAsyncFloat64Mock{
+		GaugeFunc: func(name string, opts ...instrument.Option) (asyncfloat64.Gauge, error) {
+			return nil, errors.New("some error")
+		},
+	}
+
+	fakeMeter := &fake.IMeterMock{
+		AsyncInt64Func: func() asyncint64.InstrumentProvider {
+			return fakeAsyncIntTracerProvider
+		},
+		AsyncFloat64Func: func() asyncfloat64.InstrumentProvider {
+			return fakeAsyncFloatTracerProvider
+		},
+		RegisterCallbackFunc: func(insts []instrument.Asynchronous, function func(context.Context)) error {
+			return nil
+		},
+	}
+
+	type args struct {
+		meter metric.Meter
+		mgr   client.Client
+	}
+	tests := []struct {
+		name string
+		args args
+	}{
+		{
+			name: "Basic case",
+			args: args{
+				meter: metric.NewNoopMeter(),
+				mgr:   nil,
+			},
+		},
+		{
+			name: "Error case",
+			args: args{
+				meter: fakeMeter,
+				mgr:   nil,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			SetUpKeptnMeters(tt.args.meter, tt.args.mgr)
```

Comment:
In this case, we could verify that all meters have been registered by inspecting what is being returned by the `fakeMeter.RegisterCallbackCalls()` function

Reply:
done